RandBLAS/sparse.hh

```diff
@@ -170,29 +171,26 @@ static RNGState repeated_fisher_yates(
     for (int64_t j = 0; j < vec_len; ++j)
         vec_work[j] = j;
     std::vector<int64_t> pivots(vec_nnz);
-    typedef typename T_gen::ctr_type ctr_type;
-    ctr_type rout;
-    T_gen g;
-    RNGState out_state(init_state);
+    RNG gen;
+    auto [ctr, key] = state;
```

Comment:
What makes this kind of unpacking possible? Is it the lines in https://github.com/BallisticLA/RandBLAS/pull/28/files#diff-8025ac5321d9ef206e31f5acca1b95378b5664b26ad732dcd4c4c66349eb3ce8R78-R86?

Reply:
this is because of the conversions to std::tuple you were asking about.